packages/@aws-cdk/cdk/lib/cloudformation/fn.ts

```diff
@@ -88,6 +88,18 @@ export class FnJoin extends Fn {
      * @param listOfValues The list of values you want combined.
      */
     constructor(delimiter: string, ...listOfValues: any[]) {
```

Comment:
I think I rather we align the signature here to CloudFormation instead of be "smart" about it. `FnConcat` is a nice sugar on top that has a variadic signature.

Reply:
Also missing test

Reply:
Okay. Can do that then.